File: src/utils/PerformanceService.ts
/**
 * Service to handle performance measurements for test operations.
 * Allows starting and stopping timers to calculate duration.
 */
export class PerformanceService {
  private startTime: number = 0;
  private endTime: number = 0;

  /**
   * Starts the performance timer.
   */
  startTimer(): void {
    this.startTime = performance.now();
  }

  /**
   * Stops the performance timer.
   */
  stopTimer(): void {
    this.endTime = performance.now();
  }

  /**
   * Returns the duration in milliseconds.
   * @returns number Duration in ms
   */
  getDuration(): number {
    return this.endTime - this.startTime;
  }

  /**
   * Resets the timer values.
   */
  resetTimer(): void {
    this.startTime = 0;
    this.endTime = 0;
  }
}

File: java-selenium-crawler/src/main/java/com/crawler/utils/PerformanceService.java
package com.crawler.utils;

/**
 * Service to handle performance measurements using system nanoseconds.
 */
public class PerformanceService {
    private long startTime;
    private long endTime;

    /**
     * Starts the performance timer.
     */
    public void startTimer() {
        this.startTime = System.nanoTime();
    }

    /**
     * Stops the performance timer.
     */
    public void stopTimer() {
        this.endTime = System.nanoTime();
    }

    /**
     * Returns the duration in milliseconds.
     * @return double Duration in ms (with fractional precision)
     */
    public double getDuration() {
        return (this.endTime - this.startTime) / 1_000_000.0;
    }

    /**
     * Resets the timer.
     */
    public void resetTimer() {
        this.startTime = 0;
        this.endTime = 0;
    }
}
